### PR TITLE
Store FP context when entering secure services

### PR DESCRIPTION
**Pull request by oyvindronningstad**

### CODE REVIEW

File: subsys/spm/Kconfig [hunk 1]
```diff
@@ -34,6 +34,7 @@ menuconfig IS_SPM
 	bool "Current app is SPM"
 	default n
 	select TRUSTED_EXECUTION_SECURE
+	# depends on !FPU
```
Comment (sigvartmh) [inline subsys/spm/Kconfig:37]: I would assume this is supposed to be included?

**Review by sigvartmh — CHANGES_REQUESTED** [review]
Fixes the issues seen in samples if `FPU_SHARING` is disabled in SPM.
Lack some build system changes to include the correct file in non-secure.

```
diff --git a/subsys/nonsecure/CMakeLists.txt b/subsys/nonsecure/CMakeLists.txt
index 5137c8508..ba33a5f77 100644
--- a/subsys/nonsecure/CMakeLists.txt
+++ b/subsys/nonsecure/CMakeLists.txt
@@ -6,6 +6,7 @@

 if(CONFIG_ARM_FIRMWARE_USES_SECURE_ENTRY_FUNCS)
   zephyr_sources(secure_services_ns.c)
+  zephyr_sources(${ZEPHYR_BASE}/modules/trusted-firmware-m/src/save_fp_context.c)
   zephyr_sources_ifdef(CONFIG_SPM_NS_DEBUG_LOGS_DUMP spm_ns_logs_dump.c)

   add_dependencies(zephyr_interface spm_subimage)
```

**Comment by ioannisg** [comment]
Will take a look early tomorrow :)

File: subsys/spm/Kconfig (hunk 1 above)
Comment (oyvindronningstad) [reply subsys/spm/Kconfig:37]: Yes, but it gives a Kconfig dependency loop that I couldn't figure out.

Comment (ioannisg) [reply subsys/spm/Kconfig:37]: Do you actually need the dependency? I'd vote to not list it at all.

Btw, you can clean the default n, as well.

**Review by ioannisg — APPROVED** [review]

**Review by sigvartmh — APPROVED** [review]

File: subsys/spm/Kconfig (hunk 1 above)
Comment (tejlmand) [reply subsys/spm/Kconfig:38]: I see this was merged with:
https://github.com/nrfconnect/sdk-nrf/blob/620f42ac361877a9d15c81df320300e4b8a76a43/subsys/spm/Kconfig#L38

@oyvindronningstad can you clean it up ?

Comment (oyvindronningstad) [reply subsys/spm/Kconfig:38]: https://github.com/nrfconnect/sdk-nrf/pull/4958

**Pull request merged**